src/Paket.Core/Versioning/VersionRange.fs

```diff
@@ -161,20 +228,20 @@ type VersionRequirement =
     static member Parse text =
         if String.IsNullOrWhiteSpace text || text = "null" then VersionRequirement.AllReleases else
 
-        let prereleases = ref PreReleaseStatus.No
+        let prereleases = List<string>()
```

Comment:
packages with transitive dependencies like `>= 1.0.1-beta.1 < 1.0.1` -- which are valid and can be produced automatically -- ended up with No prerelease status, as it was overwritten by the second part
also, allowing All for transitive dependencies of Concrete prerelease packages ended up with unresolvable graphs in some cases (diamond dependencies on named prerelease)